ddtrace/internal/ci_visibility/git_client.py

```diff
@@ -150,6 +150,18 @@ def wait_for_metadata_upload_status(self):
         self._wait_for_metadata_upload()
         return self._metadata_upload_status.value  # type: ignore
 
+    @classmethod
+    def _get_git_dir(cls, cwd=None):
+        # type: (Optional[str]) -> Optional[str]
+        if cwd is None:
+            cwd = os.getcwd()
+
+        git_dir = os.path.join(cwd, ".git")
+        if not os.path.exists(git_dir):
+            return None
+
+        return git_dir
```

Comment:
I know it'd be running an extra command, but, for consistency's sake and to also address the case where the `git` CLI might be missing, I think I'd prefer a `try` around `extract_workspace_path()`, eg:
```suggestion
        try:
            return extract_workspace_path()
        except ValueError:
            return None
```

Reply:
SGTM